File: BrowserApp/tvmovieapp/src/utils/swapImage.ts
// Switches images to list or poster

export function toMovieImage(
  url: string,
  type: "poster" | "landscape" = "landscape"
) {
  if (!url) return "";

  const [path, query] = url.split("?");
  const params = new URLSearchParams(query);

  if (type === "poster") {
    params.set("location", "moviePoster");
    params.set("width", "600");
    params.set("height", "900");
  } else {
    params.set("location", "landscape");
    params.set("width", "1920");
    params.set("height", "1080");
  }

  return `${path}?${params.toString()}`;
}

File: TVOSApp/TVOSTVMovieApp/TVOSTVMovieApp/utils/SwapImage.swift
import Foundation

func toMovieImage(_ url: String, type: String = "landscape") -> String {
    var base = url
    var params: [String: String] = [:]

    if let idx = url.firstIndex(of: "?") {
        base = String(url[..<idx])
        let query = String(url[url.index(after: idx)...])
        for part in query.split(separator: "&") {
            let kv = part.split(separator: "=")
            if kv.count == 2 {
                params[String(kv[0])] = String(kv[1])
            }
        }
    }

    if type == "poster" {
        params["location"] = "moviePoster"
        params["width"] = "600"
        params["height"] = "900"
    } else {
        params["location"] = "list"
        params["width"] = "1920"
        params["height"] = "1080"
    }

    let query = params
        .map { "\($0)=\($1)" }
        .joined(separator: "&")

    return "\(base)?\(query)"
}
